File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import entity.PageResult;
import entity.Result;
import com.pinyougou.pojo.TbBrand;

import java.util.List;

public interface BrandService {

    List<TbBrand> findAll();

    PageResult findPage(int pageNum, int pageSize);

    Result insertBrand(TbBrand brand);
    TbBrand findOne(long id);

    Result updateBrand(TbBrand brand);

    void deleteBrand(long[] ids);

    PageResult search(TbBrand tbBrand, int pageNum, int pageSize);

}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    //查询所有品牌
    this.findBrand=function () {
        return $http.get("../brand/findAll.do");
    }
    //分页查询所有品牌
    this.findPage=function (page,rows) {
        return $http.post("../brand/findPage.do?page="+page+"&rows="+rows);
    }
    //修改或增加品牌
    this.insert=function (entity) {
        return $http.post("../brand/insertBrand.do",entity);
    }

    this.update=function (entity) {
        return $http.post("../brand/updateBrand.do",entity);
    }

    //根据id查询品牌
    this.findOne=function (id) {

        return $http.get("../brand/findOne.do?id="+id);
    }
    //批量删除品牌
    this.dele=function (selectIds) {
        return $http.post("../brand/deleteBrands.do",selectIds);
    }

    //按条件分页查询
    this.search=function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&size="+rows,searchEntity);
    }

    this.selectOptionList=function () {
        return $http.post("../brand/selectOptionList.do");
    }
});